Userland/Libraries/LibC/dirent.cpp

```diff
@@ -278,6 +278,9 @@ int scandir(const char* dir_name,
 
     const int size = tmp_names.size();
     auto** names = static_cast<struct dirent**>(kmalloc_array(size, sizeof(struct dirent*)));
+    if (names == nullptr) {
+        return -1;
+    }
```

Comment:
We should always set `errno` if we return a value indicating failure.

```suggestion
    if (names == nullptr) {
        errno = ENOMEM;
        return -1;
    }
```

Reply:
`errno` is set via `kmalloc_array` via `malloc`.

Reply:
> errno is set via kmalloc_array via malloc.

It doesn't, and it's not required to, according to POSIX and the C standard.

Reply:
https://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html seems to be...

Reply:
But it isn't, so unless you change that too, this change is flawed.

Reply:
Should changing the following so it set `errno` and return `NULL` instead be enought ?

https://github.com/SerenityOS/serenity/blob/9592ba3548da7c09ddfa861a28bb4517cd06c6b9/Userland/Libraries/LibC/malloc.cpp#L325

Reply:
I think I have the fix sorted out. Waiting for a run test. Will probably need to add a test for that, trying to allocate a absurd amount of memory like SIZE_MAX/2.

Reply:
Added `malloc` changes so it returns `NULL` as POSIX requires. I was hopping to be able to rely on `serenity_mmap` to set `errno` correctly, but I had to collapse all its errors to `ENOMEM` and return `NULL`.